subprojects/docs/src/docs/userguide/tasks.xml

```diff
@@ -290,52 +290,673 @@
+                            about the others soon.
+                        </para>
+                    </listitem>
+                </itemizedlist>
+
+                <para>These annotated properties mean that Gradle will skip the task if none of the source
+                    files, template engine, model data or generated files have changed since the previous
+                    time Gradle executed the task. This will often save a significant amount of time. You
+                    can learn how Gradle detects <xref linkend="sec:how_does_inc_build_work">changes later</xref>.
+                </para>
+
+                <para>This example is particularly interesting because it works with collections of source
+                    files. What happens if only one source file changes? Does the task process all the
+                    source files again or just the modified one? That depends on the task implementation.
+                    If the latter, then the task itself is incremental, but that’s a different feature to
+                    the one we’re discussing here. Gradle does help task implementers with this via its
+                    <xref linkend="incremental_tasks">incremental task inputs</xref> feature.
+                </para>
+
+                <para>Now that you have seen some of the input and output annotations in practice, let’s
+                    take a look at all the annotations available to you and when you should use them. The
+                    table below lists the available annotations and the corresponding property type you
+                    can use with each one.
+                </para>
+                <table>
+                    <title>Incremental build annotations</title>
+                    <thead>
+                        <tr>
+                            <th>Annotation</th>
+                            <th>Expected property type</th>
+                            <th>Description</th>
+                        </tr>
+                    </thead>
+                    <tr>
+                        <td><literal>@Input</literal></td>
+                        <td>Any serializable type</td>
+                        <td>A simple input value</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@InputFile</literal></td>
+                        <td><literal>File</literal>*</td>
+                        <td>A single input file (not directory)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@InputDirectory</literal></td>
+                        <td><literal>File</literal>*</td>
+                        <td>A single input directory (not file)</td>
+                    </tr>
+
+                    <tr>
+                        <td><literal>@InputFiles</literal></td>
+                        <td><literal>FileCollection</literal>*</td>
```

Comment:
You make a point. Sometime I wonder if users actually know that though. I think we can leave it as is for now.